$NetBSD: patch-ac,v 1.2 2014/12/04 04:28:36 mef Exp $

Prevent error dialog when the cursor has disappeared whilst displaying
a popup `tip', as can happen when running x11/unclutter and leave the
mouse over an item with a tip; the tip is displayed, the mouse hides,
and now pck-timer doesn't barf when the mouse reappears.
Fix from Ben Elliston, who knows much more tcl than I do.

--- tips.tcl.~1~	2001-01-24 14:22:46.000000000 +1100
+++ tips.tcl
@@ -28,13 +28,18 @@ proc show_tip { win } {
 	} else {
 		set message TIP
 	}
-	toplevel .tip
+    	if ![winfo exists .tip] {
+		toplevel .tip
+	}
 	wm overrideredirect .tip 1
 	set xy [winfo pointerxy $win]
 	set x [expr [lindex $xy 0] + 8]
 	set y [expr [lindex $xy 1] + 8]
 	wm geometry .tip +$x+$y
-	message .tip.msg -text $message -font {helvetica 10} -bg bisque -bd 1 -relief raised -aspect 300
+	if ![winfo exists .tip.msg] {
+		message .tip.msg
+	}
+	.tip.msg configure -text $message -font {helvetica 10} -bg bisque -bd 1 -relief raised -aspect 300
 	pack .tip.msg
 }
 
